src/UserRepository/Htpasswd.php

```diff
@@ -61,10 +65,12 @@ public function authenticate(string $credential, string $password = null) : ?Use
         }
         fclose($handle);
 
-        return $found
-            && password_verify($password === null ? '' : $password, $hash)
-                ? $this->generateUser($credential)
-                : null;
+        if ($found && password_verify($password === null ? '' : $password, $hash)) {
+            $this->user->setIdentity($credential);
+            $this->user->setRoles($this->getRolesFromUser($credential));
+            return $this->user;
+        }
```

Comment:
This here demonstrates why passing a factory to the repository is a better approach. If we use the same instance, we cannot have concurrent requests (ala Swoole), as the instance is in the container, and can be changed by any process running at any time.

If a factory were composed in this class instead, this would become something like:

```php
return $this->userFactory->generate($credential, $this->getRolesFromUser($credential));
```

Which would guarantee that the user generated is _unique to this request_.